Update to using pyproject.toml

edsu opened the pull request:
It seems like the Python community is moving towards using pyproject.toml files to manage package and project information. This commit adds a pyproject.toml, removes the setup.py and and adds some instructions for running the tests with poetry. The github CI action was also updated.

Florents-Tselai commented:
Thanks for this, I wanted to do this, too; 
Could you please check the failing checks? 
Also, I'd like to make the package versions as lenient as possible. 
E.g. requiring python 3.11 I think is too much.

edsu commented:
@Florents-Tselai good idea about loosening the versions and allowing older pythons. I just updated the PR to allow for pythons >= 3.9 (will that be enough, or should we go lower?). I also unpinned the dependencies to see if that will work going forward. I think I mostly just copied the requirement versions from the setup.py, but I liked the idea of leaving them looser, especially for getting newer/better versions of things as they become available. 

And I think I fixed the CI build, I had forgotten to install dependencies!
